### DISCUSSION TITLE
How to format rsx element props to separate lines

### DISCUSSION THREAD

**Discussion by dominictobias** [DioxusLabs/dioxus#2110] [Q&A] [answered]
Hi, just experimenting with my first component and in VSCode the attributes get formatted all onto one line (btw my `max_width` in `rustfmt.toml` is `80`). Also I have the Dioxus extension but explicitly calling the formatters puts them all on to one line.

output:
```rust
pub fn Button(props: ButtonProps) -> Element {
    rsx! {
        button { padding: "0.5rem", margin: "0.5rem", position: "relative", color: "red", title: props.title, {props.children} }
    }
}
```

desired:
```rust
pub fn Button(props: ButtonProps) -> Element {
    rsx! {
        button {
            padding: "0.5rem",
            margin: "0.5rem",
            position: "relative",
            color: "red",
            title: props.title,
            {props.children}
        }
    }
}
```

**Comment by ealmloff** [comment, marked as answer]
It is not possible to change the default settings in the dioxus formatter currently. https://github.com/DioxusLabs/dioxus/issues/1492 is a tracking issue for making `dx fmt` respect settings for rustfmt
